packages/broker/src/config/migration.ts

```diff
@@ -0,0 +1,155 @@
+     * 
+     * Legacy plugins are dropped. Migrating a storage plugin is not supported.
+     */
+    target.plugins = {}
+    Object.keys(source.plugins).forEach((name) => {
+        const sourceConfig = source.plugins[name]
+        if (['websocket', 'mqtt', 'publishHttp'].includes(name)) {
+            target.plugins[name] = sourceConfig
+        } else if (name === 'testnetMiner') {
+            target.plugins.brubeckMiner = {}
+        } else if (name === 'metrics') {
+            const targetConfig: any = {}
+            if (sourceConfig.consoleAndPM2IntervalInSeconds !== 0) {
+                targetConfig.consoleAndPM2IntervalInSeconds = sourceConfig.consoleAndPM2IntervalInSeconds
+            }
+            target.plugins.metrics = targetConfig
+        } else if (['legacyWebsocket', 'legacyMqtt'].includes(name)) {
+            // no-op
+        } else {
+            throw new Error(`Migration not supported for plugin: ${name}`)
+        }
+    })
+    return target as Config
+}
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export const createMigratedConfig = (source: any): Config | never => {
+    const version = getVersion(source)
+    const isTestnetConfig = (version === undefined) && (isValidConfig(source, TEST_CONFIG_SCHEMA))
+    if (isTestnetConfig) {
+        return convertTestnet3ToV1(source)
+    } else {
+        throw new Error('Unable to migrate the config')
+    }
+}
+
+/* 
+ * Creates a backup file name, derived from the given name. Ensures that no file exist with that name.
+ * - foobar.ext.backup
+ * - or foobar.ext.backup-123
+ */
+const formBackupFileName = (originalFileName: string) => {
+    const name = `${originalFileName}.backup`
+    if (!fs.existsSync(name)) {
+        return name
+    } else {
+        let suffix = 1
+        while (true) {
+            const suffixedName = `${name}-${suffix}`
+            if (!fs.existsSync(suffixedName)) {
+                return suffixedName
+            }
+            suffix++
+        }
+    }
+}
+
+export const readConfigAndMigrateIfNeeded = (fileName: string | undefined): Config | never => {
+    if (fileName === undefined) {
+        fileName = [getDefaultFile(), getLegacyDefaultFile()].find((file) => fs.existsSync(file))
+        if (fileName === undefined) {
+            // eslint-disable-next-line max-len
+            throw new Error(`Config file not found in the default location. You can run "streamr-broker-init" to generate a config file interactively, or specify the config file as argument: "streamr-broker path-to-config/file.json"`)
+        }
+    }
+    let content = JSON.parse(fs.readFileSync(fileName, 'utf8'))
+    if (needsMigration(content)) {
+        const backupFile = formBackupFileName(fileName)
+        // eslint-disable-next-line no-console
+        console.log(`Migrating config ${fileName}, saving backup to ${backupFile}`)
+        fs.renameSync(fileName, backupFile)
```

Comment:
Fixed. The permissions are copied ok. Also, there were no need to explicitly set the file permissions of the new file as we just write rewrite the content instead of creating a new file.